config/persistence.php

```diff
@@ -18,4 +22,13 @@
                 tagged_iterator('.foundry.persistence.schema_resetter'),
             ])
     ;
+
+    if (PHP_VERSION_ID >= 80400) {
+        $container->services()->set('.foundry.persistence.objects_tracker', PersistedObjectsTracker::class)
+            ->tag('kernel.reset', ['method' => 'refresh'])
+            ->tag('kernel.event_listener', ['event' => TerminateEvent::class, 'method' => 'refresh'])
```

Comment:
For the record, `TerminateEvent` is needed since the kernel is only refreshed before the SECOND call to `$client->request()`